tests/settings/FunctionMetadataTest.py

```diff
@@ -107,6 +107,25 @@ def check_function_metadata_data_set(self,
         self.assertEqual(metadata.non_optional_params, non_optional_params)
         self.assertEqual(metadata.optional_params, optional_params)
 
+    def test_filter_parameters(self):
+        metadata = FunctionMetadata(
+            "A", "DESC", "RETURNS SOMETHING",
+            {"x": ("-1", None),
+             "y": ("-2", None),
+             "z": ("-3", None),
+             "w": ("-4", None)},
+            {"a": ("1", None, 1), "b": ("2", None, 2), "c": ("3", None, 3)},
+            {"y", })
+
+        self.assertEqual(
+            metadata.filter_parameters(
+                {"x": 50, "y": 60, "NOT_THERE": -100, "c": 20}),
+            {"x": 50, "c": 20})
+
+        self.assertEqual(
+            metadata.filter_parameters({"a": 50, "b": 60, "c": 70}),
+            {"a": 50, "b": 60, "c": 70})
+
```

Comment:
drop this, the doctest is sufficient and better documented and easier to understand. And less redundant.